Purchases/Identity/IdentityManager.swift

```diff
@@ -49,7 +49,11 @@ class IdentityManager {
         self.backend = backend
         self.customerInfoManager = customerInfoManager
 
-        let appUserID = appUserID
+        if appUserID?.isEmpty == true {
+            Logger.warn(Strings.identity.logging_in_with_empty_appuserid)
+        }
+
+        let appUserID = appUserID?.notEmptyOrWhitespaces
```

Comment:
I made the conscious choice to not trim whitespaces for a not-empty user ID.
For example, `" nacho with spaces "` still remains the same, but `" "` is considered empty.

Not sure why somebody would want leading or trailing whitespaces, but I figured we wouldn't want to change that behavior.